lib/auth/helpers_mfa.go

```diff
@@ -0,0 +1,328 @@
+/*
+Copyright 2017-2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth/mocku2f"
+	"github.com/gravitational/teleport/lib/auth/u2f"
+	wanlib "github.com/gravitational/teleport/lib/auth/webauthn"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/pquerna/otp"
+	"github.com/pquerna/otp/totp"
+)
+
+// TestDeviceType represents the various supported MFA types.
+type TestDeviceType int
+
+const (
+	TestOTPDevice TestDeviceType = iota
```

Comment:
I wonder if it makes sense to define device types officially in our `types.proto` file and reuse that everywhere (including [here](https://github.com/gravitational/teleport/blob/master/tool/tsh/mfa.go#L46))? Because I am creating these enums as well for `CreateRegisterChallenge`:
```
// DeviceType describes the MFA device type.
enum DeviceType {
    DEVICE_TYPE_UNSPECIFIED = 0;
    DEVICE_TYPE_TOTP = 1;
    DEVICE_TYPE_U2F = 2;
    DEVICE_TYPE_WEBAUTHN = 3;
}
```

We should've done this in beginning since https://github.com/gravitational/teleport/blob/master/api/client/proto/authservice.proto#L857, but this seems local to this request

Reply:
I wouldn't reuse the types on tsh because that is essentially UI code, so I think it's for the best for it to be separate. I'm happy to take a TODO and follow up in the next PR, is that OK?

Reply:
See https://github.com/gravitational/teleport/pull/8336